Realm/RLMObservation.mm

```diff
@@ -553,3 +555,66 @@ void RLMDidChange(std::vector<realm::BindingContext::ObserverState> const& obser
+    RLMProperty *property;
+    std::vector<std::pair<TableKey, ColKey>> keyPairs;
+
+    TableRef table = info->table();
+    NSUInteger start = 0, length = keyPath.length, end = NSNotFound;
+    do {
+        end = [keyPath rangeOfString:@"." options:0 range:{start, length - start}].location;
+        NSString *propertyName = [keyPath substringWithRange:{start, end == NSNotFound ? length - start : end - start}];
+        property = objectSchema[propertyName];
+        RLMPrecondition(property, @"Invalid property name",
+                        @"Property '%@' not found in object of type '%@'",
+                        propertyName, objectSchema.className);
+
+        if (end != NSNotFound) {
+            RLMPrecondition(property.type == RLMPropertyTypeObject || property.type == RLMPropertyTypeLinkingObjects,
+                            @"Invalid value", @"Property '%@' is not a link in object of type '%@'",
+                            propertyName, objectSchema.className);
+
+            REALM_ASSERT(property.objectClassName);
+
+            TableKey tk = table->get_key();
+            ColKey ck;
+            if (property.type == RLMPropertyTypeObject) {
+                ck = table->get_column_key(property.columnName.UTF8String);
+                table = table->get_link_target(ck);
+            } else if (property.type == RLMPropertyTypeLinkingObjects) {
+                // If the property type is a backlink, subsequent keys must be retrieved from the origin table
+                NSString *originTableName = [NSString stringWithFormat:@"class_%@", property.objectClassName];
+                StringData originTableNameStringData = RLMStringDataWithNSString(originTableName);
+                TableRef originTable = info->realm->_realm->read_group().get_table(originTableNameStringData);
+                
+                // The property from the origin table that links to the current property in the target table
+                NSString *originForwardLinkColumnName = property.linkOriginPropertyName;
+                StringData originForwardLinkColumnStringData = RLMStringDataWithNSString(originForwardLinkColumnName);
+                ColKey forwardLinkColumnKey = originTable->get_column_key(originForwardLinkColumnStringData);
+                
+                // Get the opposite (backlinked) column from the origin table
+                ck = originTable->get_opposite_column(forwardLinkColumnKey);
+                // Reassign table to the origin table so subsequent
+                // iterations can access column keys from origin table
+                table = originTable;
+            } else {
+                // This branch should never be reached. This case should have been caught by precondition above.
+                // How should an error be properly thrown?
+                RLMException(@"Precondition failure");
+            }
```

Comment:
This logic should use RLMObjectSchema and friends rather than the Table. It'll cut down on conversions between string types, and is required for renamed properties to work correctly.

Reply:
What's the proper way to get a table key and column key from an RLMObjectSchema?

Reply:
What's the proper way to get a table key and column key from an RLMObjectSchema?

Reply:
I've tried getting an realm::ObjectSchema from the RLMObjectSchema, where table_key and column_keys are accessible.
```
realm::ObjectSchema objectSchema = [rlmObjectSchema objectStoreCopy:schema];
```
except the `objectSchema.table_key` isn't the incorrect value. The `info->objectSchema.table_key` is correct though. So I think I'm looking in the wrong place.

Reply:
Keys are obtained from RLMClassInfo. `info->objectSchema.table_key` and `info->tableColumn(propertyName)`, and you get the RLMClassInfo for the target table by looking up `property.objectClassName` in the `RLMSchemaInfo` for the Realm.

Reply:
I tried to remove very use of tables directly [`9c20a19`](https://github.com/realm/realm-cocoa/pull/7249/commits/9c20a19a0db6685b1c1198eb22677080f2d3c068), except I did need to use `table->get_opposite_column` it in the newly added `realm::ColKey RLMClassInfo::computedTableColumn(RLMProperty *property) const`